PennMobile/GSR-Booking/Views/RoomCell.swift

```diff
@@ -78,9 +78,12 @@ extension RoomCell: UICollectionViewDataSource, UICollectionViewDelegate, UIColl
     // MARK: - Collection View Delegate Methods
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
         let timeSlot = room.timeSlots[indexPath.row]
-        delegate?.handleSelection(for: room, timeSlot: timeSlot, action: SelectionType.add)
-        let cell = collectionView.cellForItem(at: indexPath)
-        cell?.backgroundColor = .informationYellow
+        //if invalid time slot, then dont even handle selection
+        if timeSlot.isAvailable {
+            delegate?.handleSelection(for: room, timeSlot: timeSlot, action: SelectionType.add)
+            let cell = collectionView.cellForItem(at: indexPath)
+            cell?.backgroundColor = .informationYellow
+        }
```

Comment:
Let's use the delegate method ```func collectionView(UICollectionView, shouldSelectItemAt: IndexPath) -> Bool
``` since this use case is what it is meant for.